src/ashell/comms-uart.c

```diff
@@ -299,23 +299,40 @@ static int comms_out(int c)
 
 void comms_write_buf(const char *buf, int len)
 {
-    if (len == 0)
-        return;
+    if (comms_get_echo_mode()) {
```

Comment:
I'm a little concerned about this because write_buf is called ~20 times; I hope none of those other ones are supposed to be getting printed even though we 've turned off "echo"?